CONTRIBUTING.md

```diff
@@ -0,0 +1,188 @@
+
+We use the following labels to categorize issues:
+- *Bug*: Indicates a problem with the current implementation that needs to be fixed.
+- *Feature Request*: Suggests a new feature or enhancement to be added to the project.
+- *Documentation*: Relates to improvements or issues with project documentation.
+
+Click the green "Submit new issue" button to create the issue when you are ready.
+
+### Making changes
+Contributing to the project is easy! Follow these steps to make changes and submit them for review:
+
+#### Fork the repository
+If you still did not fork the repository, follow the instruction in [Getting Started](#getting-started).
+
+#### Create a branch
+Create a new branch to work on your changes using the following command:
+
+```bash
+git checkout -b my-feature
+```
+
+Replace "my-feature" with a descriptive name for your feature or fix.
+
+#### Make changes
+Make the necessary changes to the code, documentation, or other project files.
+
+#### Commit changes
+Once you've made your changes, commit them to your local repository using the following commands:
+
+```bash
+git add . # or be explicit and add only the changes you really want to push
+git commit -m "Add new feature"  # Replace with a descriptive commit message
+```
+
+#### Push changes
+Push your changes to your GitHub repository:
+
+```bash
+git push origin my-feature
+```
+
+#### Submit a pull request
+Go to your fork of the repository on GitHub and click the "New Pull Request" button. Compare the changes you made in 
+your branch with the main branch of the original repository.
+Provide a title and description for your pull request, detailing the changes you made. Submit your changes for review.
+
+#### Review and iterate
+A project maintainer will review your pull request and provide feedback. Make any requested changes and push them to 
+your branch. The pull request will be updated automatically.
+
+#### Merge pull request
+Once your pull request is approved, a project maintainer will merge your changes into the main branch of the repository.
+
+### Code style
+This code adheres to the [PEP 8 style guide](https://peps.python.org/pep-0008/), which promotes readability and 
+consistency in Python code:
+
+- Functions are defined using snake_case convention, where lowercase words are separated by an underscore.
+- Function names should be descriptive and reflect the purpose or functionality of the function.
+- Comments start with # and are used to explain code functionality.
+- Docstrings (multiline strings enclosed in triple quotes) are used to document non-private functions (optionally also private ones).
+- Indentation is consistent and consists of four spaces per level.
+- if `__name__ == "__main__":` block is used to allow the script to be executed directly as well as imported as a module.
+- Private functions are defined using `_name_of_private_function`.
```

Comment:
> - Private functions are defined using `_name_of_private_function`.

Proposal:
Private functions are defined starting with an underscore `_` for .e.g., `_name_of_private_function`